cpp/src/parquet/stream_reader.cc

```diff
@@ -168,28 +264,67 @@ void StreamReader::ReadFixedLength(char* ptr, int len) {
 void StreamReader::Read(ByteArray* v) {
   const auto& node = nodes_[column_index_];
   auto reader = static_cast<ByteArrayReader*>(column_readers_[column_index_++].get());
+  int16_t def_level;
+  int16_t rep_level;
   int64_t values_read;
 
-  reader->ReadBatch(1, nullptr, nullptr, v, &values_read);
+  reader->ReadBatch(1, &def_level, &rep_level, v, &values_read);
 
   if (values_read != 1) {
-    throw ParquetException("Failed to read value for column '" + node->name() + "'");
+    ThrowReadFailedException(node);
   }
 }
 
+bool StreamReader::ReadOptional(ByteArray* v) {
+  const auto& node = nodes_[column_index_];
+  auto reader = static_cast<ByteArrayReader*>(column_readers_[column_index_++].get());
+  int16_t def_level;
+  int16_t rep_level;
+  int64_t values_read;
+
+  reader->ReadBatch(1, &def_level, &rep_level, v, &values_read);
```

Comment:
please, comment constant parameters i.e.: /* parameter_name= */1

Reply:
Created a constant for this: kBatchSizeOne